src/lib/resetPasswordService.ts

```diff
@@ -0,0 +1,70 @@
+import { sign, verify } from 'jsonwebtoken';
+import { AuthistOptions, User } from './authist';
+import { getSecret } from './credentialsService';
+import { ERROR_CODE, NotAuthenticated } from './error';
+import { hashPassword } from './providers/emailPasswordProvider';
+import { handleError } from './providerUtils';
+
+const RESET_TOKEN_TYPE = 'resetPasswordToken';
+
+export const resetPassword = (options: AuthistOptions) => async (email: string, req?: any) => {
+    try {
+        const opts = options.emailPassword!;
+        const user = await opts.getUserByEmail(email, req);
+        if (!user) {
+            throw new NotAuthenticated(ERROR_CODE.UserNotFound);
+        }
+        const token = await getResetToken(options, user);
+        await (opts?.sendResetPasswordEmail ? opts.sendResetPasswordEmail(token, user, req) : Promise.resolve());
+        await (opts.saveResetPasswordToken ? opts.saveResetPasswordToken(token, user, req) : Promise.resolve());
```

Comment:
Good I can choose whether save it or not by omitting the saveResetPasswordToken :+1: